R/authenticate_sb.R

```diff
@@ -43,13 +43,39 @@ authenticate_sb = function(username, password){
 		stop('Unable to authenticate to SB. Check username and password')
 	}
 	
+	token_url <- pkg.env$token_url
+	
+	token <- POST(token_url, 
+								body = list(
+									client_id = pkg.env$keycloak_client_id,
+									grant_type = "password",
+									username = username,
+									password = password
+								), encode = "form")
+	
+	if(!token$status_code == 200) {
+		
+		warning('Unable to authenticate to SB cloud. Standard login is available.')
+		
+	} else {
+	
+		set_keycloak_env(token)
+		
+	}
+	
 	attributes(h) <- c(attributes(h), list(birthdate=Sys.time()))
 	pkg.env$session  = h
 	pkg.env$username = username
 	
 	invisible(h)
 }
 
+set_keycloak_env <- function(token_resp) {
+	pkg.env$keycloak_token <- jsonlite::fromJSON(rawToChar(token_resp$content))
+	
+	pkg.env$keycloak_expire <- Sys.time() + pkg.env$keycloak_token$expires_in
+}
+
```

Comment:
This json parsed into an object in the active environment seemed like a reasonable way to handle this.